### PR TITLE
Convert BTPreferredPaymentMethods, BTPreferredPaymentMethodsResult, BTCardNetwork and BTViewControllerPresentingDelegate

### PR DESCRIPTION
**Pull request by jaxdesmarais**
This is it, we have arrived at the final conversion files. From here, it's deleting the Objective-C module, cleanup, and general tidying but _this is it for implementation_. 🥳 

### Summary of changes

- Convert `BTPreferredPaymentMethods`, `BTPreferredPaymentMethodsResult`, `BTCardNetwork` and `BTViewControllerPresentingDelegate` to Swift
- Delete the Obj-C files
- Delete `UnitTests/BraintreeTestShared/BraintreeTestShared.h` that's no longer needed
- Update imports based on the above changes in a few test files

### Checklist

- [x] Added a changelog entry

- @jaxdesmarais

### CODE REVIEW

File: Sources/BraintreeCard/BTCardNonce.m [hunk 1]
```diff
@@ -74,7 +74,7 @@ + (NSString *)typeStringFromCardNetwork:(BTCardNetwork)cardNetwork {
             return @"Solo";
         case BTCardNetworkSwitch:
             return @"Switch";
-        case BTCardNetworkUKMaestro:
+        case BTCardNetworkUkMaestro:
```
Comment (jaxdesmarais) [inline Sources/BraintreeCard/BTCardNonce.m:77]: When converting the `BTCardNetwork` enum to Swift there is not a way to give the Obj-C case a different name than the Swift case. Due to how Obj-C maps enum names to Swift this worked and gave us `.ukMaestro` in Swift. there is not an equivalent to this mapping in Swift so this case name changed and was added to the CHANGELOG.

**Review by jcnoriega — APPROVED** [review]

File: Sources/BraintreeCoreSwift/BTPreferredPaymentMethodResult.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import Foundation
+
+///  Contains information about which payment methods are preferred on the device.
+///  This class is currently in beta and may change in future releases.
+@objcMembers public class BTPreferredPaymentMethodsResult: NSObject {
```
Comment (scannillo) [inline Sources/BraintreeCoreSwift/BTPreferredPaymentMethodResult.swift:5]:
```suggestion
///  :nodoc: Contains information about which payment methods are preferred on the device.
///  This class is currently in beta and may change in future releases.
@objcMembers public class BTPreferredPaymentMethodsResult: NSObject {
```

File: Sources/BraintreeCoreSwift/BTPreferredPaymentMethods.swift [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    var application: AnyObject
+
+    private let apiClient: BTAPIClient
+
+    // MARK: - Initializer
+
+    /// Creates an instance of BTPreferredPaymentMethods.
+    /// - Parameter apiClient: An API client
+    @objc(initWithAPIClient:)
+    public init(apiClient: BTAPIClient) {
```
Comment (scannillo) [inline Sources/BraintreeCoreSwift/BTPreferredPaymentMethods.swift:20]: We should probably add `nodoc` to all of the funcs on this class to be safe

**Review by scannillo — APPROVED** [review]
This LGTM! A few nitpicks re: adding `nodoc` to all functions in PreferredPaymentMethods files

File: Sources/BraintreeCoreSwift/BTPreferredPaymentMethods.swift [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+                result.isPayPalPreferred = false
+                result.isVenmoPreferred = isVenmoInstalled
+
+                let errorEvent: String = "ios.preferred-payment-methods.\(error != nil ? "api-error" : "api-disabled")"
+                self.apiClient.sendAnalyticsEvent(errorEvent)
+                completion(result)
+                return
+            }
+
+            let parameters: [String: Any] = ["query": "query PreferredPaymentMethods { preferredPaymentMethods { paypalPreferred } }"]
```
Comment (scannillo) [inline Sources/BraintreeCoreSwift/BTPreferredPaymentMethods.swift:58]: Side note - but there has been re-investment in getting this to work, on a new team called "Payment Intelligence" which is exciting! Hopefully we can expose it someday soon 🥳

File: Sources/BraintreeCoreSwift/BTPreferredPaymentMethods.swift (hunk 3 above)
Comment (jaxdesmarais) [reply Sources/BraintreeCoreSwift/BTPreferredPaymentMethods.swift:21]: Updated here: https://github.com/braintree/braintree_ios/pull/890/commits/57cdffb3685c054df240490b628e53b925549b72

**Pull request merged**